src/lib/Libifl/auth.c

```diff
@@ -1089,6 +1091,50 @@ auth_exec_socket(int sock, unsigned short port, char *auth_method, char* jobid)
 			return INTERACTIVE_AUTH_RETRY;
 		}
 		free_auth_config(auth_config);
+	} else if ((strcmp(auth_method, AUTH_GSS_NAME) == 0)) {
+		char encrypt_method[MAXAUTHNAME + 1] = "GSS";
+		pbs_auth_config_t *auth_config = NULL;
+		auth_def_t *authdef = NULL;
+		struct hostent *hostp;
+		const void *addr;
+		size_t addr_size;
+
+		if (load_auths(AUTH_CLIENT)) {
+			fprintf(stderr, "qsub: Failed to load auths\n");
+			return INTERACTIVE_AUTH_FAILED;
+		}
+
+		auth_config = make_auth_config(auth_method,
+					       encrypt_method,
+					       pbs_conf.pbs_exec_path,
+					       pbs_conf.pbs_home_path,
+					       NULL);
+		if (auth_config == NULL) {
+			fprintf(stderr, "qsub: Out of memory when allocating new auth config\n");
+			return INTERACTIVE_AUTH_FAILED;
+		}
+
+		authdef = get_auth(auth_method);
+		if (authdef == NULL) {
+			fprintf(stderr, "qsub: Auth method '%s' does not seem implemented\n", auth_method ? auth_method : "");
+			free_auth_config(auth_config);
+			return INTERACTIVE_AUTH_FAILED;
+		}
+
+		addr = &from->sin_addr;
+		addr_size = sizeof(from->sin_addr);
+		hostp = gethostbyaddr(addr, addr_size, from->sin_family);
```

Comment:
Why not using get_hostname_from_addr()?

Reply:
Done. Did not know the function is there. I will try to remember the function:-)